### ISSUE TITLE
Error: the "path" argument must be of type string or an instance of Buffer or URL (workaround available)

### ISSUE THREAD

**Issue by Sebert81** [alberti42/obsidian-import-attachments-plus#11]
Hi,

suddenly an error popped up.
Drag & dropping the same file from the same directory into a canvas works, however any file from the same directory (or any directory for that matter= into any note produces this error.
Worked yesterday, and I havent changed my config.

Console:

node:fs:1583 Uncaught (in promise) TypeError: The "path" argument must be of type string or an instance of Buffer or URL. Received undefined
    at stat (node:fs:1583:10)  
    at t.stat (node:electron/js2c/node_init:2:5513)  
    at node:internal/util:432:7  
    at new Promise (<anonymous>)  
    at Object.stat (node:internal/util:418:12) 
    at It (plugin:import-attachments-plus:7:5056)  
    at eval (plugin:import-attachments-plus:7:5529)  
    at Array.map (<anonymous>)  
    at Qe (plugin:import-attachments-plus:7:5509)  
    at Fe.handleFiles (plugin:import-attachments-plus:7:49625)  
    at Fe.editor_drop_cb (plugin:import-attachments-plus:7:49474)  
    at t.tryTrigger (app.js:1:723011)  
    at t.trigger (app.js:1:722944)  
    at t.trigger (app.js:1:2372770)  
    at e.handleDrop (app.js:1:1169047)  
    at Object.drop (app.js:1:1625584)  
    at app.js:1:375704  
    at e.runHandlers (app.js:1:373760)  
    at e.handleEvent (app.js:1:373522)

**Comment by alberti42** [comment]
Ok. Did you try to drag a folder into the canvas. I did not test this case since long time... The error is related to this piece of code:

```
async handleFiles(files: File[], editor: Editor, view: MarkdownView, doForceAsking: boolean, importType: ImportOperationType) {

		const {nonFolderFilesArray, foldersArray} = await Utils.filterOutFolders(Array.from(files));

		if(foldersArray.length>0) {
			const modal = new FolderImportErrorModal(this, foldersArray);
			modal.open();
			await modal.promise;
		}
```

which calls `filterOutFolders` to find out if you also have selected folders. I need to check it.

Any feedback is appreciated, especially
- if the error also occurs for files,
- if you have found a way to reproduce the error in a systematic manner

**Comment by Sebert81** [comment]
Hi,
no it happens when I drag a single file (as mentioned in my first post), not a folder, into a note.
It works when your plugin is deactivated.
I tried several file types and directories. Always the same error.

And what is weird: At home it works, but on my school laptop it doesnt.
Both PCs are synched via Obsidian official sync.

**Comment by alberti42** [comment]
Is it a windows, mac or linux laptop where it does not work? Do you have the same settings on both devices where it works and where it does not work.

I will for sure be able to find the right fix. Your feedback can speed things up.

**Comment by Sebert81** [comment]
Windows 10 on both devices. All settings,plugins etc are synced.

I also encountered an error a few minutes ago when I restarted Obsidian, it was gone when I relaunched it again.

Maybe it is relevant?

app.js:1 TypeError: Cannot convert undefined or null to object
    at Function.entries (<anonymous>)
    at te (plugin:import-attachments-plus:7:19798)
    at mt (plugin:import-attachments-plus:7:20066)
    at eval (plugin:import-attachments-plus:7:39659)
    at t.<anonymous> (app.js:1:2338022)
    at app.js:1:237228
    at Object.next (app.js:1:237333)
    at a (app.js:1:236051)

**Comment by alberti42** [comment]
I think I know what the problem is about the error you reported in the first post. Apparently, Obsidian behaves differently depending on what the installer version is. I believe the installer version is the version when you first installed Obsidian. Then Obsidian does internal updates but it does not change the installer part.

I did a new fresh installation of Obsidian and now I have both versions indicates as 1.7.6. Before, the installer version was for me 1.6.3. You can compare the two screenshots:

![Bildschirmfoto 2024-11-17 um 17 41 36](https://github.com/user-attachments/assets/09e11ba4-4332-4f10-b272-2e19430949a7)

![Bildschirmfoto 2024-11-17 um 17 43 45](https://github.com/user-attachments/assets/e34091a1-f3d5-4add-a60d-ddd5b2f5eecc)

Very unfortunately, it seems that the newer Obsidian installer does strip away information about the original path of the imported files. I need to understand how to get to this information because without the path of the file the user drags and drop into the Obsidian editor, I cannot ensure move operations.

The whole point of my plugin was to enable move operations. Then it grew to offer more functionality, but I don't want to give up to move operations. So I will have to research further. It is quite frustrating what they are doing in Obsidian and the fact that a move operation is not offered in Obisidian out of the box.

**Comment by alberti42** [comment]
I think a simple workaround is to download an old Obsidian version, for example, release https://github.com/obsidianmd/obsidian-releases/releases/tag/v1.6.3 which I know to work for sure and then let Obsidian update to the latest version. In this way, you can use the plugin and benefit from the functionalities as in the latest version of Obsidian.

Of course, I will have to dig deeper to understand why the new Obsidian stripped away information about the full path of the files dragged into the editor.

It would be great to hear back from you if this fixes the problem.

**Comment by alberti42** [comment]
I believe I could fix the problem with the newer release https://github.com/alberti42/obsidian-import-attachments-plus/releases/tag/1.5.12.

**Comment by alberti42** [comment]
Thanks for reporting on the problem. It was quite insidious because it required me to reinstall Obsidian in order to reproduce it. It was almost by chance that I found it out because I would not have thought that Obsidian does not update its framework (the Electron app framework) but only the JavaScript part.

I close this issue because I think it was resolved in release 1.5.12. Please feel free to reopen it if problems persist.

**Comment by Sebert81** [comment]
Thank you. It works again.

And indeed: At home I had an installation made with an old installer, at work I had to reinstall Obsidian.
Never thought that it would cause such a problem and thus never included it in the bug report.
